### PR TITLE
Gitlab Integration

### PR DESCRIPTION
**Pull request by valendesigns**

### CODE REVIEW

File: check-diff.sh [hunk 1]
```diff
@@ -69,6 +69,9 @@ function set_environment_variables {
 			DIFF_BASE=${DIFF_BASE:-$TRAVIS_COMMIT^}
 		fi
 		DIFF_HEAD=${DIFF_HEAD:-$TRAVIS_COMMIT}
+	elif [[ ! -z "${GITLAB_CI}" ]]; then
+		DIFF_BASE=${DIFF_BASE:-master}
```
Comment (westonruter) [inline check-diff.sh]: Let's update this to use `$DEFAULT_BASE_BRANCH` from #229.

File: check-diff.sh [hunk 2]
```diff
@@ -396,6 +399,13 @@ function install_tools {
 		DEV_LIB_SKIP="$DEV_LIB_SKIP,composer"
 	fi
 
+	# Config npm for GitLab.
+	if [[ ! -z "${GITLAB_CI}" ]]; then
+		npm config set prefix $TEMP_DIRECTORY
+		export NODE_PATH=$TEMP_DIRECTORY/lib/node_modules:$NODE_PATH
+		export PATH=$TEMP_DIRECTORY/bin:$PATH
+	fi
```
Comment (westonruter) [inline check-diff.sh]: This can be eliminated once #252 is resolved.

**Comment by PatelUtkarsh** [comment]
Should we add `.gitlab-ci.yml` sample file?  @valendesigns

**Comment by valendesigns** [comment]
@PatelUtkarsh I don't think we have enough experience with Gitlab CI to suggest a best practice sample config. Once IBD is 100% sorted I think we can circle back and add one though.

**Pull request merged**